### PR TITLE
Catch connection data that crashes proxy

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

File: sensible-proxy_test.go [hunk 1]
```diff
@@ -22,17 +23,9 @@ func (w *FakeWriter) Write(p []byte) (n int, err error) {
...
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	actual, err := requestDomainHTTP(conn, "google.com")
-	if err != nil {
-		t.Errorf("Error on connect: %s", err)
-	}
+	actual, conn, _ := requestHTTP("google.com")
+	defer conn.Close()
```
Comment (halkyon) [inline sensible-proxy_test.go:28]: could this defer be too late? could it be bundled into `requestHTTP`?

Comment (halkyon) [inline sensible-proxy_test.go:28]: you seem to be doing it for `requestHTTPS`, so maybe make this consistent.
https://github.com/mateusz/sensible-proxy/pull/5/files#diff-2c3c704edfdf526f60eaa169c62db24bR148

Comment (stojg) [inline sensible-proxy_test.go:28]: We need a open conn for `conn.LocalAddr().String()` to check the logging, so we can't close it in requestHTTP()

Comment (halkyon) [inline sensible-proxy_test.go:28]: you are deferring the close in `requestHTTPS`, would that not mean it breaks the logging for https requestS?

Comment (halkyon) [inline sensible-proxy_test.go]: oh, maybe not. Forget what I said

Comment (stojg) [inline sensible-proxy_test.go:28]: fixed

**Pull request merged**